app/src/main/kotlin/io/intrepid/skotlinton/screens/example1/Example1Activity.kt

```diff
@@ -1,29 +1,51 @@
 package io.intrepid.skotlinton.screens.example1
 
+import android.arch.lifecycle.ViewModel
 import android.content.Intent
-
+import android.widget.Toast
 import butterknife.OnClick
 import io.intrepid.skotlinton.R
 import io.intrepid.skotlinton.base.BaseMvpActivity
-import io.intrepid.skotlinton.base.PresenterConfiguration
+import io.intrepid.skotlinton.base.ViewModelConfiguration
 import io.intrepid.skotlinton.screens.example2.Example2Activity
+import io.reactivex.rxkotlin.plusAssign
 
-class Example1Activity : BaseMvpActivity<Example1Contract.Presenter>(), Example1Contract.View {
+class Example1Activity : BaseMvpActivity<Example1ViewModel>() {
 
     override val layoutResourceId: Int = R.layout.activity_example1
 
-    override fun createPresenter(configuration: PresenterConfiguration): Example1Contract.Presenter {
-        return Example1Presenter(this, configuration)
+    override val viewModelClass: Class<out ViewModel> = Example1ViewModel::class.java
+
+    override fun createViewModel(configuration: ViewModelConfiguration): Example1ViewModel {
+        return Example1ViewModel(configuration)
+    }
+
+    override fun onStart() {
+        super.onStart()
+        onStopDisposable += viewModel.eventPublisher.subscribe {
+            when (it) {
+                is Example1ViewEvent.GotoExample2 -> gotoExample2()
+                is Example1ViewEvent.ShowCurrentTimeToast -> showToast(it.message)
+            }
+        }
     }
 
-    @Suppress("ProtectedInFinal")
-    @OnClick(R.id.example1_button)
-    protected fun onButtonClick() {
-        presenter.onButtonClick()
+    @OnClick(R.id.next_button)
+    fun onButtonClick() {
```

Comment:
I like these better as `protected` than `public`, even with the suppression. On a recent project I used the pattern of adding `@Suppress("ProtectedInFinal")` at the class level for things with Butterknife bindings.